src/components/Filters.tsx

```diff
@@ -0,0 +1,311 @@
+import {
+	Checkbox,
+	FormControl,
+	FormControlLabel,
+	FormGroup,
+	FormLabel,
+	Radio,
+	RadioGroup,
+	Rating,
+	Slider,
+	styled,
+} from '@mui/material'
+import React, { Dispatch, SetStateAction, useState, useEffect } from 'react'
+import { ProductCardProps } from './ProductCard'
+import StarIcon from '@mui/icons-material/Star'
+import { green } from '@mui/material/colors'
+import { exampleProducts } from '~/data/exampleProducts'
+import { FaPercent } from 'react-icons/fa'
+import useStore from '~/store/useStore'
+
+const StyledRating = styled(Rating)({
+	'& .MuiRating-iconFilled': {
+		color: '#16a34a',
+		filter: 'drop-shadow(1px 0.75px 0px rgb(0 0 0 / 1))',
+	},
+	'& .MuiRating-iconEmpty': {
+		color: '#71717a',
+		fill: 'black',
+	},
+})
+
+interface FiltersProps {
+	searchTerm: string
+	setFilteredProducts: Dispatch<SetStateAction<ProductCardProps[]>>
+}
+
+function valuetext(value: number) {
+	return `${value}`
+}
+
+const minDistance = 500
+
+const Filters = ({ setFilteredProducts, searchTerm }: FiltersProps) => {
+	const [price, setPrice] = useState<number[]>([50, 5000])
+	const [checkedOnSale, setCheckedOnSale] = useState(false)
+	const [checkedShowedUnavailable, setCheckedShowedUnavailable] = useState(false)
+	const [checkedRating, setCheckedRating] = useState<string | number>('Any')
+	const { reducerState } = useStore()
+
+	useEffect(() => {
+		let productsToFilter = [...exampleProducts]
+
+		productsToFilter = productsToFilter.filter(product => product.price >= price[0] && product.price <= price[1])
+
+		!checkedShowedUnavailable && (productsToFilter = productsToFilter.filter(product => product.available === true))
+
+		checkedOnSale && (productsToFilter = productsToFilter.filter(product => product.onSale))
+
+		checkedRating !== 'Any' &&
+			(productsToFilter = productsToFilter.filter(product => product.rating >= Number(checkedRating)))
+
+		reducerState.activeCategory  &&
+			(productsToFilter = productsToFilter.filter(product => product.categoryId === reducerState.activeCategory))
+
+		reducerState.activeSubCategory  &&
+			(productsToFilter = productsToFilter.filter(product => product.subCategoryId === reducerState.activeSubCategory))
+
+		searchTerm &&
+			(productsToFilter = productsToFilter.filter(product =>
+				product.name.toLowerCase().includes(searchTerm.toLowerCase())
+			))
+
+		setFilteredProducts(productsToFilter)
+	}, [price, checkedOnSale, checkedShowedUnavailable, checkedRating, searchTerm, setFilteredProducts, reducerState.activeCategory, reducerState.activeSubCategory])
```

Comment:
Few things: 
1. You call the filter method for every single case, which is not optimal. You probably should call this method once.
2. This effect is triggered by any state update, so the better option here would be to memorize these results